### ISSUE TITLE
mTLS is not working without kube-proxy

### ISSUE THREAD

**Issue by dntosas** [istio/istio#35531]
### Bug Description

Deploying the same setup of Istio into two clusters with CIlium CNI:
- one with kube-proxy in place 
- one without kube-proxy (eBPF)

[Authentication](https://istio.io/latest/docs/tasks/security/authentication/authn-policy/) or [Authorization](https://istio.io/latest/docs/tasks/security/authorization/authz-dry-run/) Tasks produce different results because mTLS is not able to work, workloads can only work on PERMISSIVE mode to actually have plain communications.

Is this an intended behavior? Does mTLS somehow rely on kube-proxy?

### Version

```prose
$ i version
client version: 1.11.3
control plane version: 1.11.3
data plane version: 1.11.3 (4 proxies)
```

### Affected product area

- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [X] Security
- [ ] Test and Release
- [X] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### Is this the right place to submit this?

- [X] This is not a security vulnerability
- [X] This is not a question about how to use Istio

**Comment by dntosas** [comment]
Ok, seems related to https://github.com/istio/istio/issues/27619 and https://github.com/istio/istio/issues/32231 :/

**Comment by howardjohn** [comment]
Yeah this sounds like https://github.com/istio/istio/issues/27619. The fix is not to tweak the services, etc, but to configure cilium differently. There is a setting to keep the kube-proxy removal but also work with Istio by doing the socket translation at a later point (after Istio interception). I am not sure what the option is though - we should probably find it and document it on istio.io

**Comment by dntosas** [comment]
thanks @howardjohn for your response, let me also communicate this with Cilium maintainers to take feedback ^^

if we got this working by configure Cilium itself, I will open a PR to document this

**Comment by brb** [comment]
The cilium-agent option is named `--bpf-lb-sock-hostns-only`. You need to set it to `true` in order to prevent the cilium's socket LB from translating service connections. Otherwise, Istio sidecar proxy won't be able to determine the original destination. The PR has been merged (https://github.com/cilium/cilium/pull/17154/), just it hasn't been released yet (it will be in v1.10.5 and v1.11.0).

**Comment by JimmyCYJ** [comment]
Shall we close this in favor of #27619?

**Comment by alfsch** [comment]
For all who have troubles with cilium in strict mode without kube-proxy.

From my point of view, the root cause for the issues was our cilium version < 12.x and linux kernel < 5.7 due to the usage of ubuntu 20.04 as kubernetes node image.

There were the same issues as described in http(s) communication across namespaces using services with port mappings to different destination ports. Using the same ports mitigated the problem for some pods, but not for all.
e.g. Java based service worked well, but python services with async lib had troubles.....

After updating to cilium 1.12.6 we still faced the issues, but after looking into the logs of the cilium pods, I found in the startup section warnings about possible problems with istio using a kernel lower than 5.7.

We updated the cluster nodes to a current ubuntu 22.04 with kernel 5.15.x and voilà, all our issues were gone. We even don't need the port mapping change to use the same destination ports.

The for us working command line to render the cilium chart is:

```
   helm template cilium cilium/cilium --version 1.12.6 --namespace` kube-system --set cni.exclusive=false --set operator.replicas=1 \
     --set hubble.tls.auto.method=cronJob --set hubble.relay.enabled=true --set hubble.ui.enabled=true \
     --set socketLB.hostNamespaceOnly=true --set nodePort.enabled=true --set hostPort.enabled=true \
     --set externalIPs.enabled=true --set bpf.masquerade=false \
     --set kubeProxyReplacement=strict --set k8sServiceHost=CHANGEME --set k8sServicePort=CHANGEME  
```
kubernetes node:

```
System Info:
  Kernel Version:             5.15.0-60-generic
  OS Image:                   Ubuntu 22.04.1 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.6.16
  Kubelet Version:            v1.22.16
```
